Sources/Segment/Plugins/Platforms/Vendors/AppleUtils.swift

```diff
@@ -238,16 +238,15 @@ internal class MacOSVendorSystem: VendorSystem {
     }
     
     override var userAgent: String? {
-        var userAgent: String?
-        if Thread.isMainThread {
-            userAgent = WKWebView().value(forKey: "userAgent") as? String
-        } else {
-            DispatchQueue.main.sync {
-              userAgent = WKWebView().value(forKey: "userAgent") as? String
+        // BKS: It was discovered that on some platforms there can be a delay in retrieval.
+        // It has to be fetched on the main thread, so we've spun it off
+        // async and cache it when it comes back.
+        if Self.asyncUserAgent == nil {
+            DispatchQueue.main.async {
+                Self.asyncUserAgent = WKWebView().value(forKey: "userAgent") as? String
```

Comment:
Same question in here: Should we check the `asyncUserAgent == nil` condition here again?

Reply:
I don't mind if it gets set twice. It's locked via the @atomic marker on the var so no race condition on double-set.